Add page about password hashing algorithms (#9)
RustCrypto/book: src/key-derivation/README.md

# Key Derivation Functions

[Key derivation functions] (KDFs) provide a way to take an input such as a password, and create an
output of specified length. This output can then be used, for example, as the key for an encryption
algorithm.

> 
> Password hashing functions are a subset of key derivation functions; see

Reviewer: ```suggestion

Password hashing functions are a related concept to key derivation functions; see
```

# Key Derivation Functions

> [Key derivation functions] provide a way to take an input such as a password, and crate an output of

Reviewer 2: crate->create

Reviewer 2: Do we want to introduce the KDF acronym and use it?

Author: Thanks, fixed the typo and added the acronym

> specified length. This output can then be used as e.g. a key for an encryption algorithm.

Reviewer 2: Can you revise the sentence to not use e.g.?  Perhaps: "used, for example, as the key in"

Author: I reworded, please take another look

Reviewer 2: looks good, thanks!
